extensions/gnome/desktop-gnome-specific

```diff
@@ -28,4 +28,8 @@ if [ "$SNAP_DESKTOP_COMPONENTS_NEED_UPDATE" = "true" ]; then
   fi
 fi
 
+if [ "$SNAP_DESKTOP_COMPONENTS_NEED_UPDATE" = "true" ]; then
```

Comment:
`SNAP_DESKTOP_COMPONENTS_NEED_UPDATE` is tested in `desktop-init` but only updated in `desktop-gnome-specific`, so if using only `desktop-init` and `desktop-common` as parts of the command chain `$SNAP_USER_DATA/.last_revision` will never be updated. The result is that mime cache generation and other first-time initialization procedures will run at every execution, not only after an update. I think this should be updated in the common part of the desktop scripts.

Reply:
check changeset e6e6503c. This has the downside that if the user kills the launch scripts before completion it could leave them in an inconsistent state where the `.last-revision` file is marked as current but the full caching has not been completed due to the aborted launch.

Reply:
Hmm, yes, you're right. So perhaps we could update it at the end of `desktop-common` after the operations are done (but still in the common part)? The other possibility would be to trap early interruptions and restore the original state, but that would add unneeded complexity IMHO.